src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java

```diff
@@ -293,6 +305,9 @@ private void assignPermissions(User user,
         List<IntegrationFunctionGroupDataGroup> functionGroupDataGroups = new ArrayList<>();
 
         this.jobProfileService.getAssignedJobProfiles(externalServiceAgreementId).forEach(jobProfile -> {
+            if(isTemplate.test(jobProfile)){
+                return;
+            }
```

Comment:
It will be cleaner if we filter out unwanted items with the stream filter, like:
```
        this.jobProfileService.getAssignedJobProfiles(externalServiceAgreementId)
            .stream()
            .filter(isTemplate.negate())
            .filter(jobProfile -> jobProfileService.isJobProfileForUserRole(jobProfile, user.getRole(), isRetail))
            .forEach(jobProfile -> {

                List<String> dataGroupIds = this.productGroupService
                    .findAssignedProductGroupsIds(externalServiceAgreementId, user);
                List<IntegrationIdentifier> dataGroupIdentifiers = new ArrayList<>();

                dataGroupIds.forEach(
                    dataGroupId -> dataGroupIdentifiers.add(new IntegrationIdentifier().withIdIdentifier(dataGroupId)));

                functionGroupDataGroups.add(new IntegrationFunctionGroupDataGroup()
                    .withFunctionGroupIdentifier(
                        new IntegrationIdentifier().withIdIdentifier(jobProfile.getId()))
                    .withDataGroupIdentifiers(dataGroupIdentifiers));
            });
```

Reply:
Done.